src/libs/antares/study/parts/common/cluster_list.cpp

```diff
@@ -282,32 +278,32 @@ template<class ClusterT>
 bool ClusterList<ClusterT>::forceReload(bool reload) const
 {
     bool ret = true;
-    auto end = cluster.end();
-    for (auto i = cluster.begin(); i != end; ++i)
-        ret = (i->second)->forceReload(reload) and ret;
+    for (const auto& c : cluster)
+        ret = c->forceReload(reload) && ret;
     return ret;
 }
 
 template<class ClusterT>
 void ClusterList<ClusterT>::markAsModified() const
 {
-    auto end = cluster.end();
-    for (auto i = cluster.begin(); i != end; ++i)
-        (i->second)->markAsModified();
+    for (const auto& c : cluster)
+        c->markAsModified();
 }
 
 template<class ClusterT>
 bool ClusterList<ClusterT>::remove(const Data::ClusterName& id)
 {
-    auto i = cluster.find(id);
-    if (i == cluster.end())
+    const auto& it = find_if(cluster.begin(), cluster.end(),
```

Comment:
Here we use/call another find(...) (**std::find_if**) function.
I guess we use this one here because it gives iterator on the found cluster, that we can use when calling the std::erase function few lines further.
We should try to have only 2 **find** functions, one taking an **id**, the other a pointer to a cluster, and code the body of these function with **std::find** functions.
In that case, the caller will have to change a bit.